spec/unit/models/project_spec.rb

```diff
@@ -156,4 +156,60 @@
       end
     end
   end
+
+  describe 'transferring processes and schedules' do
+    let(:project) { double(GoodData::Project) }
+    let(:target_project) { double(GoodData::Project) }
+    let(:add_component) { double(GoodData::Process) }
+    let(:add) { double(GoodData::AutomatedDataDistribution) }
+    let(:output_stage) { double(GoodData::AdsOutputStage) }
+
+    before do
+      allow(project).to receive(:processes) { [add_component] }
+      allow(project).to receive(:add) { add }
+      allow(project).to receive(:pid)
+      allow(project).to receive(:name)
+      allow(target_project).to receive(:pid)
+      allow(target_project).to receive(:name)
+      allow(target_project).to receive(:processes) { [] }
+      allow(add_component).to receive(:component) do
+        { 'name' => 'gdc-data-distribution' }
+      end
+      allow(add_component).to receive(:name)
+      allow(add_component).to receive(:project) { project }
+      allow(add_component).to receive(:type) { :etl }
+      allow(add_component).to receive(:add_component?) { true }
+      allow(add).to receive(:output_stage) { output_stage }
+      allow(output_stage).to receive(:output_stage_prefix)
+      allow(output_stage).to receive(:schema)
+    end
+
+    describe '#transfer_processes' do
+      it 'skips ADDv2 components' do
+        subject.transfer_processes(project, target_project)
+        expect(target_project).not_to receive(:deploy_process)
+      end
+    end
+
+    describe '#transfer_schedules' do
+      let(:add_schedule) { double(GoodData::Schedule) }
+      let(:target_process) { double(GoodData::Process) }
+      let(:process_name) { 'foo process' }
+      before do
+        allow(project).to receive(:schedules).and_return([add_schedule])
+        allow(target_project).to receive(:schedules).and_return([])
+        allow(add_schedule).to receive(:to_hash).and_return(params: {})
+        allow(add_schedule).to receive(:trigger_id)
+        allow(add_schedule).to receive(:obj_id).and_return('add_schedule_id')
+        allow(target_project).to receive(:processes) { [target_process] }
+        allow(target_process).to receive(:name) { process_name }
+      end
+
+      # TODO: Make this test pass. Currently, #transfer_schedules works only when processes are in sync.
```

Comment:
in sync?

Reply:
It means processes have to be the same in both projects. So we need to run #transfer_processes right before #transfer_schedules. A difference in processes could lead to weird behavior.